src/dspeed/processors/convolutions.py

```diff
@@ -206,3 +206,54 @@ def convolve_damped_oscillator(
     for i in range(len(w_in)):
         w_out[i:] += np.real(cn) * w_in[: len(w_in) - i]
         cn *= c
+
+
+@guvectorize(
+    [
+        "void(float32[:], float32[:], float32[:])",
+        "void(float64[:], float64[:], float64[:])",
+    ],
+    "(n),(m),(p)",
+    **nb_kwargs(
+        forceobj=True,
+    ),
+)
+def reflected_convolve_wf(
+    w_in: np.ndarray, kernel: np.array, w_out: np.ndarray
+) -> None:
+    """
+    Parameters
+    ----------
+    w_in
+        the input waveform.
+    kernel
+        the kernel to convolve with
+    w_out
+        the filtered waveform.
+    """
+
+    w_out[:] = np.nan
+
+    if np.isnan(w_in).any():
+        return
+
+    if np.isnan(kernel).any():
+        return
+
+    if len(kernel) > len(w_in):
+        raise DSPFatal("The filter is longer than the input waveform")
+
+    extension_length = int(len(kernel) / 2) + 1
+
+    reflected_front = np.flip(w_in[0:extension_length])
+    reflected_end = np.flip(w_in[-extension_length:])
+
+    # Extend the signal
+
+    extended_signal = w_in
+    extended_signal = np.concatenate((extended_signal, reflected_end), axis=None)
+    extended_signal = np.concatenate((reflected_front, extended_signal), axis=None)
```

Comment:
I would suggest just using https://numpy.org/doc/stable/reference/generated/numpy.pad.html. This will simplify the code and reduce the number of array allocations.